packages/utils/telemetry-utils/src/telemetryEventBatcher.ts

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { truncateToDecimalPlaces } from "./mathTools.js";
+import type {
+	ITelemetryGenericEventExt,
+	ITelemetryLoggerExt,
+	ITelemetryPerformanceEventExt,
+} from "./telemetryTypes.js";
+
+/**
+ * Expected type of the custom data passed into the logger.
+ */
+// type Data<TKey extends string> = { readonly [key in TKey]: number };
+
+/**
+ * Expected type of the custom data passed into the logger.
+ */
+interface IMeasuredCodeResult<TKey extends string> {
+	telemetryProperties?: { readonly [key in TKey]: number };
+}
+
+/**
+ * @remarks The logger is expected to be used for a single event type. If the set of `telmetryProperties` is different for different events, a separate `TelemetryEventBatcher` should be created for each event type.
+ * Telemetry class that accumulates user defined telemetry metrics {@link ICustomDataMap} and sends it to the {@link  ITelemetryLoggerExt} logger provided to this class every time the {@link TelemetryEventBatcher.log} the number of calls to this function reaches a number specified by {@link TelemetryEventBatcher.threshold}.
+ * @typeparam TMetrics - The set of keys that should be logged.
+ * E.g., `keyof Foo` for logging properties `bar` and `baz` from `type Foo = { bar: number, baz: number }`.
+ *
+ * @internal
+ */
+export class TelemetryEventBatcher<TMetrics extends string> {
+	/**
```

Comment:
Each of these comments should be directly applied to their individual properties. As it is, all of these docs are associated just with `codeDuration`.